SwrveCommonSDK/src/main/java/com/swrve/sdk/messaging/view/SwrveDialog.java

```diff
@@ -71,4 +74,11 @@ private void goneAway() {
             }
         }
     }
+
+    public Activity getParentActivity() {
+        if (parentActivity != null) {
+            return parentActivity.get();
+        }
+        return null;
+    }
 }
```

Comment:
If we call setOwnerActivity(activity) in the constructor of our custom SwrveDialog, then getOwnerActivity should work.